src/server/game/Entities/Creature/Trainer.cpp

```diff
@@ -48,10 +48,17 @@ namespace Trainer
             if (!player->IsSpellFitByClassAndRace(trainerSpell.SpellId))
                 continue;
 
+            SpellInfo const* trainerSpellInfo = sSpellMgr->AssertSpellInfo(trainerSpell.SpellId);
+            if (!trainerSpellInfo)
```

Comment:
AssertSpellInfo() is supposed to never return null

Reply:
This is impossible to fail (Assert would be correct) - thanks to startup checks